src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java

```diff
@@ -75,5 +83,49 @@ public void should_move_invalid_zip_file_to_rejected_container() throws Exceptio
             .until(() -> testHelper.storageHasFile(inputContainer, destZipFilename), is(false));
 
         assertThat(testHelper.storageHasFile(rejectedContainer, destZipFilename)).isTrue();
+        assertThat(searchByName(rejectedContainer, destZipFilename)).hasSize(1);
+    }
+
+    @Test
+    public void should_create_a_snapshot_of_previously_rejected_file_if_its_sent_again() throws Exception {
+        // given
+        final int numberOfUploads = 2;
+
+        String fileName = testHelper.getRandomFilename("24-06-2018-00-00-00.test.zip");
+        filesToDeleteAfterTest.add(fileName);
+
+        // when
+        times(numberOfUploads, () -> {
+
+            testHelper.uploadZipFile(
```

Comment:
we could extract that from lambda and just fail in case method throws exception

Reply:
oh soz. seemed to forgot the essential purpose of the pr - multiple upload of same file. in that case, don't _really_ need that stream now do we? a simple for loop will suffice the purpose